File: puzzles/day 1/one.two.js
const fs = require("fs");
var elfCallorydb = []

fs.readFile('./input.txt', 'utf8',(err,data)=>{
    if(err)return console.log(err);
    const arrayOfInput = data.split("\n")
    let totalPerElf = 0;
    arrayOfInput.forEach(calloryItem => {
    if(calloryItem > 0){
        totalPerElf+=parseInt(calloryItem)
    }else {
    elfCallorydb.push(totalPerElf)
    totalPerElf=0;
    }
    });
    let elfCallsInDescendingOrder = elfCallorydb.sort((a,b)=>a < b ? 1 : -1)
    let top3 = elfCallsInDescendingOrder.shift() + elfCallsInDescendingOrder.shift() + elfCallsInDescendingOrder.shift()
    console.log(top3)
})

File: puzzles/day 1/python/one.two.py

elfCallorydb = []
CalloryPerElfdb = []
with open("../input.txt", 'r') as f:
    input = f.readlines()
    elfCallorydb = input

totalPerElf = 0
for value in elfCallorydb:
    if value != '\n':
        totalPerElf += int(value.replace('\n', ''))
    else:
        CalloryPerElfdb.append(totalPerElf)
        totalPerElf = 0
CalloryPerElfdb.sort(reverse=True)
topthree = CalloryPerElfdb[0] + CalloryPerElfdb[1] + CalloryPerElfdb[2]
print(topthree)
